### PR TITLE
Add reproduction for internal/protobuf issue

### PR DESCRIPTION
**Pull request by fubhy**
## Summary

- Add a regression test for negative protobuf `int64` values.
- Encode negative signed integers as ten-byte, 64-bit two's-complement varints.
- Use the package import for the internal OTLP encoder so repository typechecking succeeds.

### `unstable-distributed-otlp-protobuf-negative-int64-varint`

Module: `internal/protobuf`

Negative protobuf `int32` and `int64` values must use ten-byte two's-complement varints.

Reproduction command:

```text
pnpm test --run packages/effect/test/unstable/observability/OtlpSerialization.test.ts
```

Closes EFF-311

### CODE REVIEW

File: packages/effect/test/unstable/observability/OtlpSerializationNegativeInt64.test.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { assert, it } from "@effect/vitest"
```
Comment (tim-smart) [inline packages/effect/test/unstable/observability/OtlpSerializationNegativeInt64.test.ts]: Remove this test file and merge with the main one.
